### PR TITLE
BICAWS-2064 Test Standing Data API

### PR DESCRIPTION
**Pull request by donnyhyon**
This PR is to add the new Standing Data API and replace the old method of downloading a spreadsheet and converting to JSON. 
- Added npm command `npm run download-standing-data` and add to github actions to download new data each week
- Introduce zod to safeguard our types
- Create an index file that follows similar patterns to the `pnld-download` index file. We had to chunk the API calls as we seemed to be hitting a rate limit. We had to disable a eslint rule as the implementation we are using doesn't fall under the `no-await-in-loop` rule and the docs said it was safe to do so - [no-await-in-loop](https://eslint.org/docs/latest/rules/no-await-in-loop
)

### CODE REVIEW

File: src/Standing-data-gateway/apiConfig.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { v4 as uuidv4 } from "uuid"
+
+export const liveApiUrl =
+  "https://pss.clouddev.online/cld_StandingDataReferenceService/service/sdrs/sdrs/sdrsApi"
```
Comment (bjpirt) [inline src/Standing-data-gateway/apiConfig.ts:4]: These should be passed in as an environment variable

File: src/Standing-data-gateway/apiConfig.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+export const devApiUrl =
+  "https://crime-reference-data-api.staging.service.justice.gov.uk/cld_StandingDataReferenceService/service/sdrs/sdrs/sdrsApi"
+
+export const mojOffenceBody = {
```
Comment (bjpirt) [inline src/Standing-data-gateway/apiConfig.ts:9]: I'm not sure these should live in the config - they're more like part of the API calls

File: src/Standing-data-gateway/getApplicationRequest.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import https from "https"
+import axios from "axios"
+import * as fs from "fs"
+import { applicationRequestBody, devApiUrl } from "./apiConfig"
+
+const fileCreatedNotification = () => console.log("file created")
+
+const getAppliationRequest = () => {
```
Comment (bjpirt) [inline src/Standing-data-gateway/getApplicationRequest.ts:8]: Typo

File: src/Standing-data-gateway/getApplicationRequest.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import https from "https"
+import axios from "axios"
+import * as fs from "fs"
+import { applicationRequestBody, devApiUrl } from "./apiConfig"
+
+const fileCreatedNotification = () => console.log("file created")
```
Comment (bjpirt) [inline src/Standing-data-gateway/getApplicationRequest.ts:6]: Not sure we need this

File: src/Standing-data-gateway/apiConfig.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import { v4 as uuidv4 } from "uuid"
```
Comment (bjpirt) [inline src/Standing-data-gateway/apiConfig.ts:1]: General point - the folder name is capitalised which is inconsistent with the others

File: src/Standing-data-gateway/getApplicationRequest.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import https from "https"
```
Comment (bjpirt) [inline src/Standing-data-gateway/getApplicationRequest.ts:1]: Is this file actually used?

File: src/Standing-data-gateway/index.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  const allApiResponses = []
+  for (let i = 0; i < alphaChar.length; i += chunks) {
+    let tempArray = []
+    tempArray = alphaChar.slice(i, i + chunks)
+    const promisedTasks = tempArray.map(getOffence)
+    // eslint-disable-next-line no-await-in-loop
+    allApiResponses.push(await Promise.all(promisedTasks))
+  }
+
+  const offenceCodes = allApiResponses.flat().map((responses) => convertApiResponse(responses))
```
Comment (bjpirt) [inline src/Standing-data-gateway/index.ts:19]: Remember what I said about not needing to wrap a map function?

File: src/standing-data-gateway/index.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    tempArray = alphaChar.slice(i, i + chunks)
+    const promisedTasks = tempArray.map(getOffence)
+    // eslint-disable-next-line no-await-in-loop
+    allApiResponses.push(await Promise.all(promisedTasks))
+  }
+
+  console.log(allApiResponses)
+  const offenceCodes = allApiResponses.flat().map(convertApiResponse)
+  const data = consistentSort(offenceCodes.flat())
+  await fs.promises.appendFile(
```
Comment (bjpirt) [inline src/standing-data-gateway/index.ts:22]: Not sure you want to append the file - wouldn't we be overwriting it?

**Review by C-gyorfi — APPROVED** [review]
LGTM 🚀

**Pull request merged**